libs/ledger-live-common/src/hw/index.ts

```diff
@@ -42,7 +42,12 @@ export type TransportModule = {
 };
 const modules: TransportModule[] = [];
 export const registerTransportModule = (module: TransportModule): void => {
-  modules.push(module);
+  const index = modules.findIndex((m) => m.id === module.id);
+  if (index > -1) {
+    modules[index] = module;
+  } else {
+    modules.push(module);
+  }
```

Comment:
i would prefer not to do this (see first comment) or if we want to go to such approach, the function should instead return a "unregister" function and it's user land responsability not to register twice.